shell/platform/windows/lifecycle_plugin.cc

```diff
@@ -0,0 +1,44 @@
+// Copyright 2013 The Flutter Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+#include "flutter/shell/platform/windows/lifecycle_plugin.h"
+
+#include "flutter/shell/platform/common/client_wrapper/include/flutter/string_message_codec.h"
+
+namespace flutter {
+
+namespace {
+
+constexpr char kChannelName[] = "flutter/lifecycle";
+
+constexpr char kStateResumed[] = "AppLifecycleState.resumed";
+constexpr char kStateInactive[] = "AppLifecycleState.inactive";
+constexpr char kStatePaused[] = "AppLifecycleState.paused";
+constexpr char kStateDetached[] = "AppLifecycleState.detached";
+
+}  // namespace
+
+LifecyclePlugin::LifecyclePlugin(BinaryMessenger* messenger)
+    : channel_(std::make_unique<flutter::BasicMessageChannel<std::string>>(
+          messenger,
+          kChannelName,
+          &StringMessageCodec::GetInstance())) {}
```

Comment:
Could you explain why you chose to create a new `StringMessageCodec` instead of using the `StandardMessageCodec` instead?

Reply:
Ah never mind, this is necessary to be compatible with Flutter's existing [`SystemChannel`](https://github.com/flutter/flutter/blob/526ca0d498120e93efbdd25f8824825dd7b6b13f/packages/flutter/lib/src/services/system_channels.dart#L258-L261).